spark/src/test/scala/org/apache/spark/sql/delta/DeltaProtocolVersionSuite.scala

```diff
@@ -2056,6 +2056,121 @@ trait DeltaProtocolVersionSuiteBase extends QueryTest
     }
   }
 
+  test("error message with protocol too high - table path") {
+    withTempDir { path =>
+      spark.range(1).write.format("delta").save(path.getCanonicalPath)
+      val log = DeltaLog.forTable(spark, path)
+      log.ensureLogDirectoryExist()
+
+      var tableReaderVersion = 4
+      var tableWriterVersion = 7
+      overwriteDeltaLogWithVersion(log, tableReaderVersion, tableWriterVersion)
+
+      val exceptionRead = intercept[InvalidProtocolVersionException] {
+        spark.read.format("delta").load(path.getCanonicalPath)
+      }
+      assert(exceptionRead.getMessage == getExpectedProtocolErrorMessage(
+        log.dataPath.toString,
+        tableReaderVersion,
+        tableWriterVersion))
+
+      tableReaderVersion = 3
+      tableWriterVersion = 8
+      overwriteDeltaLogWithVersion(log, tableReaderVersion, tableWriterVersion)
+      val exceptionWrite = intercept[InvalidProtocolVersionException] {
+        spark.range(1).write
+          .mode("append")
+          .option("mergeSchema", "true")
+          .format("delta")
+          .save(path.getCanonicalPath)
+      }
+      assert(exceptionWrite.getMessage == getExpectedProtocolErrorMessage(
+        log.dataPath.toString,
+        tableReaderVersion,
+        tableWriterVersion))
+    }
+  }
+
+  test("error message with protocol too high - table name") {
+    val protocolTableName = "mytableprotocoltoohigh"
+    withTable(protocolTableName) {
+      spark.range(1).write.format("delta").saveAsTable(protocolTableName)
+      val log = DeltaLog.forTable(spark, TableIdentifier(protocolTableName))
+      log.ensureLogDirectoryExist()
+
+      var tableReaderVersion = 4
+      var tableWriterVersion = 7
+      overwriteDeltaLogWithVersion(log, tableReaderVersion, tableWriterVersion)
+
+      val exceptionRead = intercept[InvalidProtocolVersionException] {
+        spark.read.format("delta").table(protocolTableName)
+      }
+
+      // During reads we can't get the table name, so we get the path instead
+      var pathInErrorMessage = log.dataPath.toString
```

Comment:
Show a long string instead of a table name isn't ideal for users, but it's nonetheless better than showing nothing. I wonder if @ryan-johnson-databricks knows any trick to get the `CatalogTable` during Snapshot reconstruction?

Reply:
The catalog table is not accessible during snapshot construction, but whoever triggers snapshot construction could catch the original exception and replace the table name?

For example, a large proportion of Delta table accesses go through [DeltaRelation.fromV2Relation](https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala#L1143), and you could make the exception a case class and do:
```diff
  def fromV2Relation(
      d: DeltaTableV2,
      v2Relation: DataSourceV2Relation,
      options: CaseInsensitiveStringMap): LogicalRelation = {
    recordFrameProfile("DeltaAnalysis", "fromV2Relation") {
+     val table = d.withOptions(options.asScala.toMap)
+     try {
+       table.initialSnapshot // trigger snapshot construction in a controlled setting
+     } catch {
+       case e: InvalidProtocolVersionException if table.catalogTable.isDefined =>
+         throw e.copy(tableNameOrPath = table.name)
+     }
+     val relation = d.toBaseRelation
-     val relation = d.withOptions(options.asScala.toMap).toBaseRelation
      val output = if (CDCReader.isCDCRead(options)) {
        // Handles cdc for the spark.read.options().table() code path
        relation.schema.toAttributes
      } else {
        v2Relation.output
      }
      LogicalRelation(relation, output, d.ttSafeCatalogTable, isStreaming = false)
    }
  }
```

Aside: In an ideal world, [DeltaCatalog.loadTable](https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala#L188) would be a single narrow waist for all accesses, but we can't safely access the snapshot there because it may not be the final version of the table. Time travel and hadoop options could still be applied later on, which can affect which snapshot we create and also whether we're even able to access the storage location. I'm trying to clean up that situation, but it will take time.

Reply:
Actually, `DeltaTableV2.name` is not safe -- it accesses `DeltaLog.dataPath` which in turn requires a snapshot to exist, and we already know we failed to create a snapshot. Better to do instead:
```scala
case e: InvalidProtocolVersionException if table.catalogTable.isDefined =>
  throw e.copy(tableNameOrPath = table.catalogTable.get.identifier)
```

Reply:
I was doing experiments by adding the try/catch/rethrow here: https://github.com/delta-io/delta/blob/5f9b98e86590cf8891c73eb2b78482aee89547d1/spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala#L88

I'll test your suggestion

Reply:
Also, is it preferable to rethrow the copy of exception, or add the original exception as initCause?

Reply:
> I was doing experiments by adding the try/catch/rethrow

Yeah... that will only create a snapshot the first time a given table is accessed on a cluster so probably not giving the results you hope for. We need to get to a part of the code that actually creates the snapshot.

> I'll test your suggestion

Keep in mind, that spot is _not_ giving 100% coverage, tho it should cover ~all read path queries and a majority of DML as well, if I'm not mistaken. DDL commands will likely need separate treatment because they use `ResolvedTable` instead of `DataSourceV2Relation`

> is it preferable to rethrow the copy of exception, or add the original exception as initCause?

Do whatever you think is most idiomatic for scala/java.

Reply:
It is actually not calling `fromV2Relation`, the error happens before it, this is the stack trace:

```
[info] - error message with protocol too high - table name *** FAILED ***
[info]   org.apache.spark.sql.delta.InvalidProtocolVersionException: [DELTA_INVALID_PROTOCOL_VERSION] Delta protocol version is not supported by this version of Delta Lake: table "file:/sources/spark/spark-warehouse/org.apache.spark.sql.delta.DeltaProtocolVersionSuite/mytableprotocoltoohigh" requires reader version 4 and writer version 7, client supports reader versions 0, 1, 2, 3 and writer versions 0, 1, 2, 3, 4, 5, 7. Please upgrade to a newer release.
[info]   at org.apache.spark.sql.delta.DeltaLog.protocolCheck(DeltaLog.scala:417)
[info]   at org.apache.spark.sql.delta.DeltaLog.protocolRead(DeltaLog.scala:451)
[info]   at org.apache.spark.sql.delta.Snapshot.init(Snapshot.scala:170)
[info]   at org.apache.spark.sql.delta.Snapshot.<init>(Snapshot.scala:419)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$createSnapshot$2(SnapshotManagement.scala:377)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotFromGivenOrEquivalentLogSegment(SnapshotManagement.scala:496)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotFromGivenOrEquivalentLogSegment$(SnapshotManagement.scala:484)
[info]   at org.apache.spark.sql.delta.DeltaLog.createSnapshotFromGivenOrEquivalentLogSegment(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshot(SnapshotManagement.scala:370)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshot$(SnapshotManagement.scala:363)
[info]   at org.apache.spark.sql.delta.DeltaLog.createSnapshot(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$createSnapshotAtInitInternal$1(SnapshotManagement.scala:324)
[info]   at scala.Option.map(Option.scala:230)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotAtInitInternal(SnapshotManagement.scala:321)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotAtInitInternal$(SnapshotManagement.scala:318)
[info]   at org.apache.spark.sql.delta.DeltaLog.createSnapshotAtInitInternal(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$getSnapshotAtInit$1(SnapshotManagement.scala:313)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.DeltaLog.recordFrameProfile(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.getSnapshotAtInit(SnapshotManagement.scala:308)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.getSnapshotAtInit$(SnapshotManagement.scala:307)
[info]   at org.apache.spark.sql.delta.DeltaLog.getSnapshotAtInit(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$init$(SnapshotManagement.scala:57)
[info]   at org.apache.spark.sql.delta.DeltaLog.<init>(DeltaLog.scala:79)
[info]   at org.apache.spark.sql.delta.DeltaLog$.$anonfun$apply$4(DeltaLog.scala:817)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:323)
[info]   at org.apache.spark.sql.delta.DeltaLog$.$anonfun$apply$3(DeltaLog.scala:812)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.DeltaLog$.recordFrameProfile(DeltaLog.scala:636)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:133)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:128)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:117)
[info]   at org.apache.spark.sql.delta.DeltaLog$.recordOperation(DeltaLog.scala:636)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:132)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:122)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:112)
[info]   at org.apache.spark.sql.delta.DeltaLog$.recordDeltaOperation(DeltaLog.scala:636)
[info]   at org.apache.spark.sql.delta.DeltaLog$.createDeltaLog$1(DeltaLog.scala:811)
[info]   at org.apache.spark.sql.delta.DeltaLog$.$anonfun$apply$5(DeltaLog.scala:829)
[info]   at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4739)
[info]   at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3524)
[info]   at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2317)
[info]   at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2280)
[info]   at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2195)
[info]   at com.google.common.cache.LocalCache.get(LocalCache.java:3934)
[info]   at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4736)
[info]   at org.apache.spark.sql.delta.DeltaLog$.getDeltaLogFromCache$1(DeltaLog.scala:828)
[info]   at org.apache.spark.sql.delta.DeltaLog$.apply(DeltaLog.scala:838)
[info]   at org.apache.spark.sql.delta.DeltaLog$.forTable(DeltaLog.scala:715)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.deltaLog$lzycompute(DeltaTableV2.scala:89)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.deltaLog(DeltaTableV2.scala:88)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.$anonfun$initialSnapshot$3(DeltaTableV2.scala:139)
[info]   at scala.Option.getOrElse(Option.scala:189)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.initialSnapshot$lzycompute(DeltaTableV2.scala:139)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.initialSnapshot(DeltaTableV2.scala:120)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.$anonfun$tableSchema$2(DeltaTableV2.scala:162)
[info]   at scala.Option.getOrElse(Option.scala:189)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.tableSchema$lzycompute(DeltaTableV2.scala:162)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.tableSchema(DeltaTableV2.scala:160)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.schema(DeltaTableV2.scala:167)
[info]   at org.apache.spark.sql.connector.catalog.Table.columns(Table.java:65)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.columns(DeltaTableV2.scala:56)
[info]   at org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation$.create(DataSourceV2Relation.scala:195)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$createRelation$1(Analyzer.scala:1251)
[info]   at scala.Option.map(Option.scala:230)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.createRelation(Analyzer.scala:1223)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$resolveRelation$3(Analyzer.scala:1274)
[info]   at scala.Option.orElse(Option.scala:447)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$resolveRelation$1(Analyzer.scala:1272)
[info]   at scala.Option.orElse(Option.scala:447)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.org$apache$spark$sql$catalyst$analysis$Analyzer$ResolveRelations$$resolveRelation(Analyzer.scala:1264)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$$anonfun$apply$14.applyOrElse(Analyzer.scala:1127)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$$anonfun$apply$14.applyOrElse(Analyzer.scala:1091)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsUpWithPruning$3(AnalysisHelper.scala:138)
[info]   at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:104)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsUpWithPruning$1(AnalysisHelper.scala:138)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:323)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsUpWithPruning(AnalysisHelper.scala:134)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsUpWithPruning$(AnalysisHelper.scala:130)
[info]   at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperatorsUpWithPruning(LogicalPlan.scala:31)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.apply(Analyzer.scala:1091)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.apply(Analyzer.scala:1050)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$2(RuleExecutor.scala:222)
[info]   at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
[info]   at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
[info]   at scala.collection.immutable.List.foldLeft(List.scala:91)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1(RuleExecutor.scala:219)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1$adapted(RuleExecutor.scala:211)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.execute(RuleExecutor.scala:211)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.org$apache$spark$sql$catalyst$analysis$Analyzer$$executeSameContext(Analyzer.scala:228)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$execute$1(Analyzer.scala:224)
[info]   at org.apache.spark.sql.catalyst.analysis.AnalysisContext$.withNewAnalysisContext(Analyzer.scala:173)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:224)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:188)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$executeAndTrack$1(RuleExecutor.scala:182)
[info]   at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:88)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.executeAndTrack(RuleExecutor.scala:182)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$executeAndCheck$1(Analyzer.scala:209)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.markInAnalyzer(AnalysisHelper.scala:330)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:208)
[info]   at org.apache.spark.sql.execution.QueryExecution.$anonfun$analyzed$1(QueryExecution.scala:76)
[info]   at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:111)
[info]   at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$2(QueryExecution.scala:202)
[info]   at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:526)
[info]   at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:202)
[info]   at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)
[info]   at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:201)
[info]   at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:76)
[info]   at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:74)
[info]   at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:66)
[info]   at org.apache.spark.sql.Dataset$.$anonfun$ofRows$1(Dataset.scala:90)
[info]   at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)
[info]   at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:88)
[info]   at org.apache.spark.sql.DataFrameReader.table(DataFrameReader.scala:608)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$2(DeltaProtocolVersionSuite.scala:2104)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
[info]   at org.apache.spark.sql.test.SQLTestUtilsBase.withTable(SQLTestUtils.scala:306)
[info]   at org.apache.spark.sql.test.SQLTestUtilsBase.withTable$(SQLTestUtils.scala:304)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuite.withTable(DeltaProtocolVersionSuite.scala:3165)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$1(DeltaProtocolVersionSuite.scala:2096)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   at org.apache.spark.SparkFunSuite.withFixture(SparkFunSuite.scala:207)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:224)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:236)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:236)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:218)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterEach$$super$runTest(SparkFunSuite.scala:66)
[info]   at org.scalatest.BeforeAndAfterEach.runTest(BeforeAndAfterEach.scala:234)
[info]   at org.scalatest.BeforeAndAfterEach.runTest$(BeforeAndAfterEach.scala:227)
[info]   at org.apache.spark.SparkFunSuite.runTest(SparkFunSuite.scala:66)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTests$1(AnyFunSuiteLike.scala:269)
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTests(AnyFunSuiteLike.scala:269)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTests$(AnyFunSuiteLike.scala:268)
[info]   at org.scalatest.funsuite.AnyFunSuite.runTests(AnyFunSuite.scala:1564)
[info]   at org.scalatest.Suite.run(Suite.scala:1114)
[info]   at org.scalatest.Suite.run$(Suite.scala:1096)
[info]   at org.scalatest.funsuite.AnyFunSuite.org$scalatest$funsuite$AnyFunSuiteLike$$super$run(AnyFunSuite.scala:1564)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$run$1(AnyFunSuiteLike.scala:273)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.run(AnyFunSuiteLike.scala:273)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.run$(AnyFunSuiteLike.scala:272)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterAll$$super$run(SparkFunSuite.scala:66)
[info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[info]   at org.apache.spark.SparkFunSuite.run(SparkFunSuite.scala:66)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   at java.lang.Thread.run(Thread.java:750)
```

Reply:
@ryan-johnson-databricks as fromV2Relation won't work, I proceeded with the change in DeltaTableV2.deltaLog, this is an example of exception:

<details>
[info] - error message with protocol too high - table name *** FAILED ***
[info]   org.apache.spark.sql.delta.InvalidProtocolVersionException: [DELTA_INVALID_PROTOCOL_VERSION] Delta protocol version is not supported by this version of Delta Lake: table "default.mytableprotocoltoohigh" requires reader version 4 and writer version 7, client supports reader versions 0, 1, 2, 3 and writer versions 0, 1, 2, 3, 4, 5, 7. Please upgrade to a newer release.
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.liftedTree1$1(DeltaTableV2.scala:107)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.deltaLog$lzycompute(DeltaTableV2.scala:94)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.deltaLog(DeltaTableV2.scala:88)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.$anonfun$initialSnapshot$3(DeltaTableV2.scala:159)
[info]   at scala.Option.getOrElse(Option.scala:189)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.initialSnapshot$lzycompute(DeltaTableV2.scala:159)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.initialSnapshot(DeltaTableV2.scala:140)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.$anonfun$tableSchema$2(DeltaTableV2.scala:182)
[info]   at scala.Option.getOrElse(Option.scala:189)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.tableSchema$lzycompute(DeltaTableV2.scala:182)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.tableSchema(DeltaTableV2.scala:180)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.schema(DeltaTableV2.scala:187)
[info]   at org.apache.spark.sql.connector.catalog.Table.columns(Table.java:65)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.columns(DeltaTableV2.scala:56)
[info]   at org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation$.create(DataSourceV2Relation.scala:195)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$createRelation$1(Analyzer.scala:1251)
[info]   at scala.Option.map(Option.scala:230)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.createRelation(Analyzer.scala:1223)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$resolveRelation$3(Analyzer.scala:1274)
[info]   at scala.Option.orElse(Option.scala:447)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$resolveRelation$1(Analyzer.scala:1272)
[info]   at scala.Option.orElse(Option.scala:447)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.org$apache$spark$sql$catalyst$analysis$Analyzer$ResolveRelations$$resolveRelation(Analyzer.scala:1264)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$$anonfun$apply$14.applyOrElse(Analyzer.scala:1127)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$$anonfun$apply$14.applyOrElse(Analyzer.scala:1091)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsUpWithPruning$3(AnalysisHelper.scala:138)
[info]   at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:104)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsUpWithPruning$1(AnalysisHelper.scala:138)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:323)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsUpWithPruning(AnalysisHelper.scala:134)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsUpWithPruning$(AnalysisHelper.scala:130)
[info]   at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperatorsUpWithPruning(LogicalPlan.scala:31)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.apply(Analyzer.scala:1091)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.apply(Analyzer.scala:1050)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$2(RuleExecutor.scala:222)
[info]   at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
[info]   at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
[info]   at scala.collection.immutable.List.foldLeft(List.scala:91)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1(RuleExecutor.scala:219)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1$adapted(RuleExecutor.scala:211)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.execute(RuleExecutor.scala:211)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.org$apache$spark$sql$catalyst$analysis$Analyzer$$executeSameContext(Analyzer.scala:228)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$execute$1(Analyzer.scala:224)
[info]   at org.apache.spark.sql.catalyst.analysis.AnalysisContext$.withNewAnalysisContext(Analyzer.scala:173)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:224)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:188)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$executeAndTrack$1(RuleExecutor.scala:182)
[info]   at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:88)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.executeAndTrack(RuleExecutor.scala:182)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$executeAndCheck$1(Analyzer.scala:209)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.markInAnalyzer(AnalysisHelper.scala:330)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:208)
[info]   at org.apache.spark.sql.execution.QueryExecution.$anonfun$analyzed$1(QueryExecution.scala:76)
[info]   at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:111)
[info]   at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$2(QueryExecution.scala:202)
[info]   at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:526)
[info]   at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:202)
[info]   at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)
[info]   at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:201)
[info]   at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:76)
[info]   at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:74)
[info]   at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:66)
[info]   at org.apache.spark.sql.Dataset$.$anonfun$ofRows$1(Dataset.scala:90)
[info]   at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)
[info]   at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:88)
[info]   at org.apache.spark.sql.DataFrameReader.table(DataFrameReader.scala:608)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$2(DeltaProtocolVersionSuite.scala:2104)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
[info]   at org.apache.spark.sql.test.SQLTestUtilsBase.withTable(SQLTestUtils.scala:306)
[info]   at org.apache.spark.sql.test.SQLTestUtilsBase.withTable$(SQLTestUtils.scala:304)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuite.withTable(DeltaProtocolVersionSuite.scala:3162)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$1(DeltaProtocolVersionSuite.scala:2096)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   at org.apache.spark.SparkFunSuite.withFixture(SparkFunSuite.scala:207)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:224)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:236)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:236)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:218)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterEach$$super$runTest(SparkFunSuite.scala:66)
[info]   at org.scalatest.BeforeAndAfterEach.runTest(BeforeAndAfterEach.scala:234)
[info]   at org.scalatest.BeforeAndAfterEach.runTest$(BeforeAndAfterEach.scala:227)
[info]   at org.apache.spark.SparkFunSuite.runTest(SparkFunSuite.scala:66)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTests$1(AnyFunSuiteLike.scala:269)
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTests(AnyFunSuiteLike.scala:269)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTests$(AnyFunSuiteLike.scala:268)
[info]   at org.scalatest.funsuite.AnyFunSuite.runTests(AnyFunSuite.scala:1564)
[info]   at org.scalatest.Suite.run(Suite.scala:1114)
[info]   at org.scalatest.Suite.run$(Suite.scala:1096)
[info]   at org.scalatest.funsuite.AnyFunSuite.org$scalatest$funsuite$AnyFunSuiteLike$$super$run(AnyFunSuite.scala:1564)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$run$1(AnyFunSuiteLike.scala:273)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.run(AnyFunSuiteLike.scala:273)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.run$(AnyFunSuiteLike.scala:272)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterAll$$super$run(SparkFunSuite.scala:66)
[info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[info]   at org.apache.spark.SparkFunSuite.run(SparkFunSuite.scala:66)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   at java.lang.Thread.run(Thread.java:750)
[info]   Cause: org.apache.spark.sql.delta.InvalidProtocolVersionException: [DELTA_INVALID_PROTOCOL_VERSION] Delta protocol version is not supported by this version of Delta Lake: table "file:/sources/spark/spark-warehouse/org.apache.spark.sql.delta.DeltaProtocolVersionSuite/mytableprotocoltoohigh" requires reader version 4 and writer version 7, client supports reader versions 0, 1, 2, 3 and writer versions 0, 1, 2, 3, 4, 5, 7. Please upgrade to a newer release.
[info]   at org.apache.spark.sql.delta.DeltaLog.protocolCheck(DeltaLog.scala:417)
[info]   at org.apache.spark.sql.delta.DeltaLog.protocolRead(DeltaLog.scala:451)
[info]   at org.apache.spark.sql.delta.Snapshot.init(Snapshot.scala:170)
[info]   at org.apache.spark.sql.delta.Snapshot.<init>(Snapshot.scala:419)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$createSnapshot$2(SnapshotManagement.scala:377)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotFromGivenOrEquivalentLogSegment(SnapshotManagement.scala:496)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotFromGivenOrEquivalentLogSegment$(SnapshotManagement.scala:484)
[info]   at org.apache.spark.sql.delta.DeltaLog.createSnapshotFromGivenOrEquivalentLogSegment(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshot(SnapshotManagement.scala:370)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshot$(SnapshotManagement.scala:363)
[info]   at org.apache.spark.sql.delta.DeltaLog.createSnapshot(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$createSnapshotAtInitInternal$1(SnapshotManagement.scala:324)
[info]   at scala.Option.map(Option.scala:230)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotAtInitInternal(SnapshotManagement.scala:321)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotAtInitInternal$(SnapshotManagement.scala:318)
[info]   at org.apache.spark.sql.delta.DeltaLog.createSnapshotAtInitInternal(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$getSnapshotAtInit$1(SnapshotManagement.scala:313)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.DeltaLog.recordFrameProfile(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.getSnapshotAtInit(SnapshotManagement.scala:308)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.getSnapshotAtInit$(SnapshotManagement.scala:307)
[info]   at org.apache.spark.sql.delta.DeltaLog.getSnapshotAtInit(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$init$(SnapshotManagement.scala:57)
[info]   at org.apache.spark.sql.delta.DeltaLog.<init>(DeltaLog.scala:79)
[info]   at org.apache.spark.sql.delta.DeltaLog$.$anonfun$apply$4(DeltaLog.scala:817)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:323)
[info]   at org.apache.spark.sql.delta.DeltaLog$.$anonfun$apply$3(DeltaLog.scala:812)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.DeltaLog$.recordFrameProfile(DeltaLog.scala:636)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:133)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:128)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:117)
[info]   at org.apache.spark.sql.delta.DeltaLog$.recordOperation(DeltaLog.scala:636)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:132)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:122)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:112)
[info]   at org.apache.spark.sql.delta.DeltaLog$.recordDeltaOperation(DeltaLog.scala:636)
[info]   at org.apache.spark.sql.delta.DeltaLog$.createDeltaLog$1(DeltaLog.scala:811)
[info]   at org.apache.spark.sql.delta.DeltaLog$.$anonfun$apply$5(DeltaLog.scala:829)
[info]   at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4739)
[info]   at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3524)
[info]   at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2317)
[info]   at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2280)
[info]   at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2195)
[info]   at com.google.common.cache.LocalCache.get(LocalCache.java:3934)
[info]   at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4736)
[info]   at org.apache.spark.sql.delta.DeltaLog$.getDeltaLogFromCache$1(DeltaLog.scala:828)
[info]   at org.apache.spark.sql.delta.DeltaLog$.apply(DeltaLog.scala:838)
[info]   at org.apache.spark.sql.delta.DeltaLog$.forTable(DeltaLog.scala:715)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.liftedTree1$1(DeltaTableV2.scala:95)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.deltaLog$lzycompute(DeltaTableV2.scala:94)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.deltaLog(DeltaTableV2.scala:88)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.$anonfun$initialSnapshot$3(DeltaTableV2.scala:159)
[info]   at scala.Option.getOrElse(Option.scala:189)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.initialSnapshot$lzycompute(DeltaTableV2.scala:159)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.initialSnapshot(DeltaTableV2.scala:140)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.$anonfun$tableSchema$2(DeltaTableV2.scala:182)
[info]   at scala.Option.getOrElse(Option.scala:189)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.tableSchema$lzycompute(DeltaTableV2.scala:182)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.tableSchema(DeltaTableV2.scala:180)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.schema(DeltaTableV2.scala:187)
[info]   at org.apache.spark.sql.connector.catalog.Table.columns(Table.java:65)
[info]   at org.apache.spark.sql.delta.catalog.DeltaTableV2.columns(DeltaTableV2.scala:56)
[info]   at org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation$.create(DataSourceV2Relation.scala:195)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$createRelation$1(Analyzer.scala:1251)
[info]   at scala.Option.map(Option.scala:230)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.createRelation(Analyzer.scala:1223)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$resolveRelation$3(Analyzer.scala:1274)
[info]   at scala.Option.orElse(Option.scala:447)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$resolveRelation$1(Analyzer.scala:1272)
[info]   at scala.Option.orElse(Option.scala:447)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.org$apache$spark$sql$catalyst$analysis$Analyzer$ResolveRelations$$resolveRelation(Analyzer.scala:1264)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$$anonfun$apply$14.applyOrElse(Analyzer.scala:1127)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$$anonfun$apply$14.applyOrElse(Analyzer.scala:1091)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsUpWithPruning$3(AnalysisHelper.scala:138)
[info]   at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:104)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsUpWithPruning$1(AnalysisHelper.scala:138)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:323)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsUpWithPruning(AnalysisHelper.scala:134)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsUpWithPruning$(AnalysisHelper.scala:130)
[info]   at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperatorsUpWithPruning(LogicalPlan.scala:31)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.apply(Analyzer.scala:1091)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.apply(Analyzer.scala:1050)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$2(RuleExecutor.scala:222)
[info]   at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
[info]   at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
[info]   at scala.collection.immutable.List.foldLeft(List.scala:91)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1(RuleExecutor.scala:219)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1$adapted(RuleExecutor.scala:211)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.execute(RuleExecutor.scala:211)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.org$apache$spark$sql$catalyst$analysis$Analyzer$$executeSameContext(Analyzer.scala:228)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$execute$1(Analyzer.scala:224)
[info]   at org.apache.spark.sql.catalyst.analysis.AnalysisContext$.withNewAnalysisContext(Analyzer.scala:173)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:224)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:188)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$executeAndTrack$1(RuleExecutor.scala:182)
[info]   at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:88)
[info]   at org.apache.spark.sql.catalyst.rules.RuleExecutor.executeAndTrack(RuleExecutor.scala:182)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$executeAndCheck$1(Analyzer.scala:209)
[info]   at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.markInAnalyzer(AnalysisHelper.scala:330)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:208)
[info]   at org.apache.spark.sql.execution.QueryExecution.$anonfun$analyzed$1(QueryExecution.scala:76)
[info]   at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:111)
[info]   at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$2(QueryExecution.scala:202)
[info]   at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:526)
[info]   at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:202)
[info]   at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)
[info]   at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:201)
[info]   at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:76)
[info]   at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:74)
[info]   at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:66)
[info]   at org.apache.spark.sql.Dataset$.$anonfun$ofRows$1(Dataset.scala:90)
[info]   at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)
[info]   at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:88)
[info]   at org.apache.spark.sql.DataFrameReader.table(DataFrameReader.scala:608)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$2(DeltaProtocolVersionSuite.scala:2104)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
[info]   at org.apache.spark.sql.test.SQLTestUtilsBase.withTable(SQLTestUtils.scala:306)
[info]   at org.apache.spark.sql.test.SQLTestUtilsBase.withTable$(SQLTestUtils.scala:304)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuite.withTable(DeltaProtocolVersionSuite.scala:3162)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$1(DeltaProtocolVersionSuite.scala:2096)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   at org.apache.spark.SparkFunSuite.withFixture(SparkFunSuite.scala:207)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:224)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:236)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:236)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:218)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterEach$$super$runTest(SparkFunSuite.scala:66)
[info]   at org.scalatest.BeforeAndAfterEach.runTest(BeforeAndAfterEach.scala:234)
[info]   at org.scalatest.BeforeAndAfterEach.runTest$(BeforeAndAfterEach.scala:227)
[info]   at org.apache.spark.SparkFunSuite.runTest(SparkFunSuite.scala:66)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTests$1(AnyFunSuiteLike.scala:269)
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTests(AnyFunSuiteLike.scala:269)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTests$(AnyFunSuiteLike.scala:268)
[info]   at org.scalatest.funsuite.AnyFunSuite.runTests(AnyFunSuite.scala:1564)
[info]   at org.scalatest.Suite.run(Suite.scala:1114)
[info]   at org.scalatest.Suite.run$(Suite.scala:1096)
[info]   at org.scalatest.funsuite.AnyFunSuite.org$scalatest$funsuite$AnyFunSuiteLike$$super$run(AnyFunSuite.scala:1564)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$run$1(AnyFunSuiteLike.scala:273)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.run(AnyFunSuiteLike.scala:273)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.run$(AnyFunSuiteLike.scala:272)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterAll$$super$run(SparkFunSuite.scala:66)
[info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[info]   at org.apache.spark.SparkFunSuite.run(SparkFunSuite.scala:66)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   at java.lang.Thread.run(Thread.java:750)
</details